ttnn/ttnn/core.py

```diff
@@ -335,17 +335,17 @@ def create_sharded_memory_config_(
             if shard_orientation == ttnn.ShardOrientation.ROW_MAJOR:
                 tensor_height_padded = roundup(tensor_height, grid_size.y * 32) if tile_layout else tensor_height
                 shard_shape = divup(tensor_height_padded, grid_size.y), divup(tensor_width, grid_size.x)
-
             elif shard_orientation == ttnn.ShardOrientation.COL_MAJOR:
                 tensor_height_padded = roundup(tensor_height, grid_size.x * 32) if tile_layout else tensor_height
                 shard_shape = divup(tensor_height_padded, grid_size.x), divup(tensor_width, grid_size.y)
             else:
                 raise RuntimeError("Invalid shard orientation")
         elif tensor_memory_layout == TensorMemoryLayout.HEIGHT_SHARDED:
-            tensor_height_padded = roundup(tensor_height, total_num_cores) if tile_layout else tensor_height
```

Comment:
Move this change to separate pull req. Since this change is not affecting only sweep tests.

Reply:
That would require some additional modifications to the `gen_sharded_spec_unary` function. But I will do it.